### DISCUSSION TITLE
Classification example, inference isn't working with custom model

### DISCUSSION THREAD

**Discussion by trr-reds** [bitcraze/discussions#569] [Q&A]
Hello,
We have created a simple model respecting the inputs and outputs of your classification model. The program runs correctly after the flash but no inference results are received, the application seems to be stuck on the call to "pi_cluster_send_task_to_cl" in the "cam_handler" function.

Here is the nngraph at compile time
![image](https://user-images.githubusercontent.com/114081696/219415163-9786bfc8-b11c-4e3e-916c-4c5729729526.png)

Here is the cfclient console
```
CPX: GAP8: *** Classification ***
CPX: GAP8: Starting to open camera
CPX: GAP8: Opened Camera
CPX: GAP8: Allocated memory for camera buffer
CPX: GAP8: Allocated memory for output
CPX: GAP8: Allocated memory for task
CPX: GAP8: Constructed CNN
CPX: GAP8: ------Run inference


```

And our cam_handler function where we print this last message
```c
static void cam_handler(void *arg)
{
  uint32_t resolution = CAM_WIDTH * CAM_HEIGHT;
  uint32_t captureSize = resolution * sizeof(unsigned char);

  pi_camera_control(&camera, PI_CAMERA_CMD_STOP, 0);

  cpxPrintToConsole(LOG_TO_CRTP, "------Run inference\n");
  /* Run inference */
  pi_cluster_send_task_to_cl(&cluster_dev, task);
  
  cpxPrintToConsole(LOG_TO_CRTP, "------END inference\n");

  if (Output_1[0] > Output_1[1])
  {
    cpxPrintToConsole(LOG_TO_CRTP, "Cup,     confidence: %hd\n", Output_1[0] - Output_1[1]);
  }
  else
  {
    cpxPrintToConsole(LOG_TO_CRTP, "Background, confidence: %hd\n", Output_1[1] - Output_1[0]);
  }

  pi_camera_capture_async(&camera, cameraBuffer, CAM_WIDTH * CAM_HEIGHT, pi_task_callback(&task1, cam_handler, NULL));
  pi_camera_control(&camera, PI_CAMERA_CMD_START, 0); 

  if (wifiClientConnected == 1) {
    createImageHeaderPacket(&txp, captureSize, RAW_ENCODING);
    cpxSendPacketBlocking(&txp);
    sendBufferViaCPX(&txp, cameraBuffer, captureSize);
  }
}
```

Do you have any idea why our model doesn't work?

Let me know if something is missing to understand my problem :)

**Comment by knmcguire** [comment]
I'm assuming you are using the GAP flow process from greenwaves to generate the CNN? Have you tried to contact Greenwaves for this? It's their tooling we are using.

https://github.com/GreenWaves-Technologies/gap_sdk
